### PR TITLE
implemented optional weekly email for user

### PR DESCRIPTION
**Pull request by sungmee0219**
Not sure how frontend is going to implement the optional weekly email, but I made some baby changes that will hopefully go along future dev for this feature!

### CODE REVIEW

File: db/migrate/20201117010549_add_optional_weekly_email_to_user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddOptionalWeeklyEmailToUser < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :weekly_email, :boolean, default: true
```
Comment (Naokimi) [inline db/migrate/20201117010549_add_optional_weekly_email_to_user.rb]: would change the name to `notifications_enabled` or `email_notifications_enabled`

File: script/start_new_week.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
 UserAvailability.destroy_all
 TimeBlock.update_all("time = time + interval '7' day")
-User.all.each { |user| WeeklyMailer.with(user: user).start_new_week.deliver_now }
+User.where(weekly_email: true).find_each do |user|
```
Comment (Naokimi) [inline script/start_new_week.rb]: should work also with just `.each`

**Review by Naokimi — COMMENTED** [review]
@JupitersLB please decide how you want to implement the FE, if via a rails view or React

**Comment by JupitersLB** [comment]
Was reached out to on Slack about it. I think going through rails would be best, but shouldn't need another view. It is an on/off switch, so could use the dropdown on the avatar which would be more explicit, or have an icon next to the avatar which lights up when active. I'm happy to help either way it is decided.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class UsersController < ApplicationController
+	def update
+		@user = User.find(params[:id])
+		@user.weekly_email = !@user.weekly_email
```
Comment (Naokimi) [inline app/controllers/users_controller.rb]: you call it `weekly_email` but the column is `email_notification_enabled`

File: db/migrate/20201117010549_add_optional_weekly_email_to_user.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddOptionalWeeklyEmailToUser < ActiveRecord::Migration[6.0]
```
Comment (Naokimi) [inline db/migrate/20201117010549_add_optional_weekly_email_to_user.rb]: you need to remove this migration

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class UsersController < ApplicationController
+	def update
+		@user = User.find(params[:id])
+		@user.weekly_email = !@user.weekly_email 
+		@user.save
+		# redirect_to user_root_path
```
Comment (Naokimi) [inline app/controllers/users_controller.rb]: remove

**Review by JupitersLB — APPROVED** [review]

File: script/start_new_week.rb [hunk 6]
```diff
@@ -1,4 +1,6 @@
 UserAvailability.destroy_all
 TimeBlock.update_all("time = time + interval '7' day")
 Company.first.update!(reminder_sent: false)
-User.all.each { |user| WeeklyMailer.with(user: user).start_new_week.deliver_later }
+User.where(weekly_email: true).each do |user|
```
Comment (Naokimi) [inline script/start_new_week.rb]: need to change this too :)

File: app/views/users/update.js.erb [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+if (emailButton.innerHTML === "Email Notifications On"){
+	emailButton.innerHTML = "Email Notifications Off";
+}else{
+	emailButton.innerHTML = "Email Notifications On";
+}
+
+if(notificationElement){
+	notificationElement.innerHTML = "<%= j render partial: 'shared/flashes', locals: {notice: flash[:notice]} %>";
+}else{
```
Comment (Naokimi) [inline app/views/users/update.js.erb]: I think there should be some spaces around `if` and `else`? not sure abotu JS code style best practices

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+class UsersController < ApplicationController
+	def update
+		@user = User.find(params[:id])
+		@user.email_notification_enabled = !@user.email_notification_enabled 
+		@user.save
+		respond_to do |format|
+			format.js {flash.now[:notice] = "Email Notification Changed Successfully"}
+		end
+	end
```
Comment (Naokimi) [inline app/controllers/users_controller.rb:9]: would prefer smaller spacing (2 spaces = 1 tab)

Comment (JupitersLB) [reply app/controllers/users_controller.rb:9]: spacing is fine locally. Must just be an issue with on GH or something

**Pull request merged**